addons/project_todo/static/src/web/activity/activity_menu_patch.js

```diff
@@ -42,4 +42,33 @@ patch(ActivityMenu.prototype, {
             size: "md",
         });
     },
+
+    availableViews(group) {
+        if (group.name === "To-Do") {
+            return this.todoViews;
+        }
+        return super.availableViews(group);
+    },
+
+    async loadTodoViews() {
+        this.todoViews = await this.orm.call(
+            "project.task",
+            "get_todo_views_id",
+            [],
+        );
+    },
+
+    async onClickAction(action, group) {
+        if (group.name === "To-Do") {
+            await this.loadTodoViews();
+        }
+        return super.onClickAction(...arguments);
+    },
+
+    async openActivityGroup(group, filter = "all") {
+        if (group.name === "To-Do") {
```

Comment:
Since the group name is translated is not really a good idea to do that, we could maybe add an extra key/value in the group object to know if the group is todo. 🙂